dicoogle/src/main/java/pt/ua/dicoogle/server/web/servlets/WelcomeServlet.java

```diff
@@ -0,0 +1,107 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Dicoogle.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package pt.ua.dicoogle.server.web.servlets;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Servlet for serving welcome messages for the webapp.
+ */
+public class WelcomeServlet extends HttpServlet {
+
+    private static final Logger LOG = LoggerFactory.getLogger(WelcomeServlet.class);
+
+    // thread-local random generator to pick messages randomly
+    private ThreadLocal<Random> random = new ThreadLocal<Random>() {
+        @Override
+        protected Random initialValue() {
+            return new Random();
+        }
+    };
+
+    /** A list of messages fetched from a file
+     * (embedded resource by default,
+     * overridable with JVM property `dicoogle.welcomeFile`)
+     */
+    private static String[] MESSAGES = loadMessages();
+
+    /** A welcome message for when no messages are available */
+    private static final String FALLBACK_MESSAGE = "Welcome to Dicoogle!";
+
+    private static String[] loadMessages() {
+        try {
+            List<String> welcomeFull;
+            String welcomeFile = System.getProperty("dicoogle.welcomeFile");
+            if (welcomeFile != null) {
+                welcomeFull = Files.readAllLines(Paths.get(welcomeFile));
+            } else {
+                InputStream is = WelcomeServlet.class.getResourceAsStream("/welcome.html");
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(is, "UTF-8"))) {
+                    welcomeFull = reader.lines().collect(Collectors.toList());
+                }
+            }
+            List<String> messages = new ArrayList<>();
+            String msg = "";
+            for (String line : welcomeFull) {
+                if (line.startsWith("#")) {
+                    continue;
+                }
+                // identify end of message (an empty line)
+                if (line.isEmpty()) {
+                    messages.add(msg);
+                    msg = "";
+                } else {
+                    msg += line + "\n";
+                }
+            }
+            if (!msg.isEmpty()) {
+                messages.add(msg);
+            }
+            if (!messages.isEmpty()) {
+                messages.add(FALLBACK_MESSAGE);
+            }
```

Comment:
Oopsie. This adds the fallback welcome message when the list is not empty, instead of when it is.

```suggestion
            if (messages.isEmpty()) {
                messages.add(FALLBACK_MESSAGE);
            }
```